### PR TITLE
Subaru: add flag checking helpers

### PR DESCRIPTION
**Pull request by jnewb1**
using &'s everywhere will be easy to mess up. for example

`(SubaruFlags.HYBRID | SubaruFlags.GLOBAL_GEN2) & self.CP.flags` will return true for global gen2 and hybrids

but in the future if we wanted  "gen2 hybrid", and did this:

`GEN2_HYBRID = SubaruFlags.HYBRID | SubaruFlags.GLOBAL_GEN2`

`SubaruFlags.GEN2_HYBRID & self.CP.flags` will return true for non-hybrid gen2 or hybrid non-gen2s

we instead have to do

`(SubaruFlags.GEN2_HYBRID & self.CP.flags) == SubaruFlags.GEN2_HYBRID`, which is very verbose and confusing

instead, I'm adding

`SubaruFlags.HYBRID.check(CP.flags)` -> is car hybrid
`(SubaruFlags.HYBRID | SubaruFlags.GEN2).any(CP.flags` -> is car hybrid or gen2

and 
`(SubaruFlags.HYBRID | SubaruFlags.GEN2).check(CP.flags)` will fail, since `SubaruFlags.HYBRID | SubaruFlags.GEN2` is not part of the `SubaruFlags` enum

### CODE REVIEW

File: selfdrive/car/__init__.py [hunk 1]
```diff
@@ -301,3 +301,13 @@ def create_carinfo_map(cls) -> dict[str, CarInfos]:
   @classmethod
   def with_flags(cls, flags: IntFlag) -> set['Platforms']:
     return {p for p in cls if p.config.flags & flags}
+
+
+class CarFlags(IntFlag):
+  def check(self, flags: int) -> bool:
+    if self not in [f.value for f in self.__class__]:
```
Comment (sshane) [inline selfdrive/car/__init__.py:308]:
```suggestion
    if len(self) != 1:
```

Comment (sshane) [reply selfdrive/car/__init__.py:308]: What is the difference between `SubaruFlags.GLOBAL_GEN2.any(SubaruFlags.GLOBAL_GEN2)` and `SubaruFlags.GLOBAL_GEN2.check(SubaruFlags.GLOBAL_GEN2)`?

Comment (sshane) [reply selfdrive/car/__init__.py:308]:
```python
In [5]: flags = SubaruFlags.GLOBAL_GEN2 #| SubaruFlags.HYBRID
   ...: (SubaruFlags.GLOBAL_GEN2 | SubaruFlags.HYBRID).all_in(flags)
Out[5]: False

In [6]: flags = SubaruFlags.GLOBAL_GEN2 | SubaruFlags.HYBRID
  ...: (SubaruFlags.GLOBAL_GEN2 | SubaruFlags.HYBRID).all_in(flags)
Out[6]: True
```

**Comment by jnewb1** [comment]
it just needs to be clear where you are do any *any* check or an *all* check at a glance, without having to know exactly how bit anding works

`ret.dashcamOnly = bool(ret.flags & (SubaruFlags.PREGLOBAL | SubaruFlags.LKAS_ANGLE | SubaruFlags.HYBRID))`

**Comment by jnewb1** [comment]
maybe we just ban flag *or*ing so that you have to explicitly check each

`(ret.flags & SubaruFlags.PREGLOBAL) or (...)`

**Comment by jnewb1** [comment]
I think `.all` is confusing when you only have 1 flag to check though

`SubaruFlags.GLOBAL_GEN2.all(self.CP.flags)`

**Comment by sshane** [comment]
I don't see why there needs to be two functions for this. The or is clear

**Comment by jnewb1** [comment]
this is the case I'm worried about that might make it unclear

`GEN2_HYBRID = GEN2 | HYBRID`

then try checking

```
# is the car a GEN2 hybrid
if GEN2_HYBRID & self.CP.flags:
```

File: selfdrive/car/tests/test_platform_configs.py [hunk 2]
```diff
@@ -17,5 +17,27 @@ def test_configs(self):
         self.assertIsNotNone(platform.config.specs)
 
 
+
```
Comment (jnewb1) [inline selfdrive/car/tests/test_platform_configs.py:20]:
```suggestion
```

File: selfdrive/car/__init__.py (hunk 1 above)
Comment (sshane) [reply selfdrive/car/__init__.py:325]: We don't use is_set, do we need it?

**Comment by jnewb1** [comment]
I'll think about this more, but I think as we add more and more flags it will be important to make it easily understandable

**Pull request closed without merging**